infrastructure/api_mtls.tf

```diff
@@ -0,0 +1,130 @@
+# New API Gateway for mTLS
+resource "aws_api_gateway_rest_api" "ndr_doc_store_api_mtls" {
+  name        = "${terraform.workspace}-DocStoreAPI-mTLS"
```

Comment:
There's a ticket in the backlog to start to standardise names. e.g. lambdas are like ndra_CustomLambda

So would be good to do it here, so whether this should be ${terraform.workspace}_DocStoreApiMtls  

I guess the main thing is all resources should probably begin in the same way, like the lambdas i.e. ${terraform.workspace}_ not ${terraform.workspace}- 

This would apply anywhere you have given a resource a name with the workspace